extension/otto/gohan_db.go

```diff
@@ -147,6 +200,39 @@ func init() {
 				value, _ := vm.ToValue(resp.Data())
 				return value
 			},
+			"gohan_db_lock_fetch": func(call otto.FunctionCall) otto.Value {
+				VerifyCallArguments(&call, "gohan_db_lock_fetch", 4)
+				transaction, needCommit, err := env.GetOrCreateTransaction(call.Argument(0))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				if needCommit {
+					defer transaction.Close()
+				}
+				schemaID, err := GetString(call.Argument(1))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				ID, err := GetString(call.Argument(2))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				tenantID, err := GetString(call.Argument(3))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+
+				resp, err := GohanDbLockFetch(transaction, schemaID, ID, tenantID)
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+
+				if resp == nil {
+					otto.NullValue()
```

Comment:
Nice catch.
In fact it's common code with gohan_db_fetch, and the contract of underlaying transaction. Fetch guarantees that we'd either get a result or an error. So I'd rather add a panic()